mono/metadata/reflection.c

```diff
@@ -2641,6 +2641,10 @@ ves_icall_RuntimeMethodInfo_MakeGenericMethod_impl (MonoReflectionMethodHandle r
 	MonoMethod *imethod = reflection_bind_generic_method_parameters (method, types, error);
 	return_val_if_nok (error, MONO_HANDLE_CAST (MonoReflectionMethod, NULL_HANDLE));
 
+	MonoReflectionType *reftype = MONO_HANDLE_GETVAL (rmethod, reftype);
+	g_assert (reftype->type->type == MONO_TYPE_CLASS);
```

Comment:
Is this enough for handling the typed union? Is there a scenario where this could be any other type?

Reply:
Judging by the test suite barking at me, the answer is yes